pkg/apis/pipeline/v1beta1/pipelinerun_types.go

```diff
@@ -331,6 +331,10 @@ type PipelineRunStatusFields struct {
 
 	// PipelineRunSpec contains the exact spec used to instantiate the run
 	PipelineSpec *PipelineSpec `json:"pipelineSpec,omitempty"`
+
+	// list of tasks that were skipped due to when expressions evaluating to false
+	// +optional
+	SkippedTasks []string `json:"skippedTasks,omitempty"`
```

Comment:
one of the options for skipped tasks in [the tep](https://github.com/tektoncd/community/pull/192) involves adding more information under "skipped tasks" - maybe it makes sense to make this a list of a more complex object so we can add more fields later without making a backwards incompatible change?

depends on how likely you think it is that we'll want that extra information.